File: power_of_thor/taille_min/pot_taille.py
l,m,t,u=map(int,input().split())
while input():
 d,e=(l>t)-(l<t),(m>u)-(m<u);t+=d;u+=e;print('S'*(e>0)+'N'*(e<0)+'E'*(d>0)+'W'*(d<0))

File: power_of_thor/taille_min/pot_taille.rs
use std::io::{self, Write};
fn main() {
    let mut input = String::new();
    io::stdin().read_line(&mut input).unwrap();
    let mut coords: Vec<i32> = input.split_whitespace().map(|num| num.parse().unwrap()).collect();
    let (mut c, mut d) = (coords[2], coords[3]);
    loop {
        io::stdin().read_line(&mut input).unwrap();
        let (x, y) = (coords[0], coords[1]);
        let mut move_dir = String::new();
        if d<y {move_dir.push('S'); d+=1; }
        if d>y {move_dir.push('N'); d-=1; }
        if c<x {move_dir.push('E'); c+=1; }
        if c>x {move_dir.push('W'); c-=1; }
        println!("{}", move_dir);
        io::stdout().flush().unwrap();
    }
}
